model/riscv_vlen.sail

```diff
@@ -21,9 +25,13 @@ function get_elen_pow() = match elen {
 
 register vlen : bits(4)
 
+val sys_vector_vlen_exp = pure "sys_vector_vlen_exp" : unit -> bits(4)
+
+function get_vlen() -> bits(4) = sys_vector_vlen_exp()
+
 val get_vlen_pow : unit -> {5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}
 
-function get_vlen_pow() = match vlen {
+function get_vlen_pow() = match get_vlen() {
```

Comment:
You'll need to update the code that actually uses `vlen` to call this function (similarly for `elen`). Easy to find once you remove the registers.